File: recipe-java/src/main/java/nl/rabobank/kotlinmovement/recipes/data/IngredientsEntity.java
package nl.rabobank.kotlinmovement.recipes.data;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name = "ingredients")
public class IngredientsEntity {
    @ManyToOne(cascade = CascadeType.REMOVE, optional = false)
    @JoinColumn(name = "recipes_id", nullable = false)
    private RecipesEntity recipes;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String type;
    private Integer weight;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        IngredientsEntity that = (IngredientsEntity) o;

        if (getId() == null || that.getId() == null) return false;

        return getId().equals(that.getId());
    }

    @Override
    public int hashCode() {
        if (getId() == null) {
            return super.hashCode();
        } else {
            return getId().hashCode();
        }
    }
}

File: java-to-kotlin-complete/src/main/kotlin/nl/rabobank/kotlinmovement/recipes/data/IngredientsEntity.kt
package nl.rabobank.kotlinmovement.recipes.data

import javax.persistence.*

@Entity
@Table(name = "ingredients")
class IngredientsEntity(
    @ManyToOne(optional = false)
    @JoinColumn(name = "recipes_id", nullable = false)
    val recipes: RecipesEntity? = null,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    val name: String,
    val type: String,
    val weight: Int,
) {

    override fun equals(other: Any?): Boolean {
        return when {
            other !is IngredientsEntity -> false
            this === other -> true
            id == null || other.id == null -> false
            else -> id == other.id
        }
    }

    override fun hashCode(): Int {
        return id?.hashCode() ?: super.hashCode()
    }
}
